storm-core/src/jvm/org/apache/storm/container/cgroup/CgroupCommon.java

```diff
@@ -0,0 +1,247 @@
+ * limitations under the License.
+ */
+package org.apache.storm.container.cgroup;
+
+import org.apache.storm.container.cgroup.core.CgroupCore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class CgroupCommon implements CgroupCommonOperation {
+
+    public static final String TASKS = "/tasks";
+    public static final String NOTIFY_ON_RELEASE = "/notify_on_release";
+    public static final String RELEASE_AGENT = "/release_agent";
+    public static final String CGROUP_CLONE_CHILDREN = "/cgroup.clone_children";
+    public static final String CGROUP_EVENT_CONTROL = "/cgroup.event_control";
+    public static final String CGROUP_PROCS = "/cgroup.procs";
+
+    private final Hierarchy hierarchy;
+
+    private final String name;
+
+    private final String dir;
+
+    private final CgroupCommon parent;
+
+    //private final Map<SubSystemType, CgroupCore> cores;
+
+    private final boolean isRoot;
+
+    //private final Set<CgroupCommon> children = new HashSet<CgroupCommon>();
+
+    private static final Logger LOG = LoggerFactory.getLogger(CgroupCommon.class);
+
+    public CgroupCommon(String name, Hierarchy hierarchy, CgroupCommon parent) {
+        this.name = parent.getName() + "/" + name;
+        this.hierarchy = hierarchy;
+        this.parent = parent;
+        this.dir = parent.getDir() + "/" + name;
+        this.isRoot = false;
+    }
+
+    /**
+     * rootCgroup
+     */
+    public CgroupCommon(Hierarchy hierarchy, String dir) {
+        this.name = "";
+        this.hierarchy = hierarchy;
+        this.parent = null;
+        this.dir = dir;
+        this.isRoot = true;
+    }
+
+    @Override
+    public void addTask(int taskId) throws IOException {
+        CgroupUtils.writeFileByLine(CgroupUtils.getDir(this.dir, TASKS), String.valueOf(taskId));
+    }
+
+    @Override
+    public Set<Integer> getTasks() throws IOException {
+        List<String> stringTasks = CgroupUtils.readFileByLine(CgroupUtils.getDir(this.dir, TASKS));
+        Set<Integer> tasks = new HashSet<Integer>();
+        for (String task : stringTasks) {
+            tasks.add(Integer.valueOf(task));
+        }
+        return tasks;
+    }
+
+    @Override
+    public void addProcs(int pid) throws IOException {
+        CgroupUtils.writeFileByLine(CgroupUtils.getDir(this.dir, CGROUP_PROCS), String.valueOf(pid));
+    }
+
+    @Override
+    public Set<Integer> getPids() throws IOException {
+        List<String> stringPids = CgroupUtils.readFileByLine(CgroupUtils.getDir(this.dir, CGROUP_PROCS));
+        Set<Integer> pids = new HashSet<Integer>();
+        for (String task : stringPids) {
+            pids.add(Integer.valueOf(task));
+        }
+        return pids;
+    }
+
+    @Override
+    public void setNotifyOnRelease(boolean flag) throws IOException {
```

Comment:
I understand that. But cgroup config files should one value only in the file at a time? writeFileByLine should override the file with latest value. If you read getNotifyOnRelease, you will read the first line of file which is actually the oldest. please refer to my comment on writeFileByLine